inception-recommendation-api/src/main/java/de/tudarmstadt/ukp/inception/recommendation/api/evaluation/EvaluationResult.java

```diff
@@ -17,20 +17,167 @@
+    private List<String> labels;
+
+    private int total;
+    private int numOfLabels;
+
+    // TODO replace or delete
+    private double defaultScore;
+
+    /**
+     * Calculate macro-averaged scores on per-token basis over all labels contained in the given
+     * annotated pairs except for those matching the given ignore-label.
+     * 
+     * @param aIgnoreLabel
+     *            this label will be ignored during evaluation
+     * @param aAnnotatedPairs
+     *            pairs of gold and predicted labels for the same token
+     */
+    public EvaluationResult(String aIgnoreLabel, Stream<AnnotatedTokenPair> aAnnotatedPairs)
+    {
+        super();
+        labels = new ArrayList<>();
+        ignoreLabel = aIgnoreLabel;
+        // construct confusion matrix
+        confusionMatrix = new HashMap<>();
+        aAnnotatedPairs.filter(this::isEqualIgnoreLabel).forEach(this::incConfusionMatrix);
+    }
+
+    public EvaluationResult()
     {
-        return defaultScore;
+    }
+
+    private boolean isEqualIgnoreLabel(AnnotatedTokenPair aPair)
+    {
+        return !aPair.getGoldLabel().equals(ignoreLabel)
+                && !aPair.getPredictedLabel().equals(ignoreLabel);
+    }
+
+    private void incConfusionMatrix(AnnotatedTokenPair aPair)
+    {
+        String goldLabel = aPair.getGoldLabel();
+        String predictedLabel = aPair.getPredictedLabel();
+
+        // annotated pair is true positive
+        if (goldLabel.equals(predictedLabel)) {
+            incCounter(goldLabel, goldLabel);
+        }
+        else {
+            // annotated pair is false negative for gold class = annotated pair is false
+            // positive for predicted class
+            incCounter(goldLabel, predictedLabel);
+        }
+
+        total += 1;
+    }
+
+    private void incCounter(String aGoldLabel, String aPredictedLabel)
+    {
+        if (!confusionMatrix.containsKey(aGoldLabel)) {
+            Map<String, Integer> initMap = new HashMap<>();
+            initMap.put(aGoldLabel, 0);
+            confusionMatrix.put(aGoldLabel, initMap);
+
+            labels.add(aGoldLabel);
+            numOfLabels++;
+        }
+        if (!confusionMatrix.get(aGoldLabel).containsKey(aPredictedLabel)) {
+            confusionMatrix.get(aGoldLabel).put(aPredictedLabel, 0);
+        }
+        int count = confusionMatrix.get(aGoldLabel).get(aPredictedLabel) + 1;
+        confusionMatrix.get(aGoldLabel).put(aPredictedLabel, count);
+    }
+
+    /**
+     * Calculate accuracy
+     * 
+     * @return accuracy score
+     */
+    public double getAccuracyScore()
+    {
+        double tp = 0.0;
+        for (String label : labels) {
+            tp += confusionMatrix.get(label).get(label);
+        }
+        return (total > 0) ? tp / (double) total : 0.0;
+    }
+
+    /**
+     * Calculate macro-averaged precision score
+     * 
+     * @return precision score
+     */
+    public double getPrecisionScore()
```

Comment:
I would call it `computePrecisionScore`, `get` implies that it is a cheap operation (e.g. just returning a variable value)